library/kani_macros/src/sysroot/contracts/shared.rs

```diff
@@ -197,3 +197,15 @@ pub fn try_as_result_assign(stmt: &syn::Stmt) -> Option<&syn::LocalInit> {
 pub fn try_as_result_assign_mut(stmt: &mut syn::Stmt) -> Option<&mut syn::LocalInit> {
     try_as_result_assign_pat!(stmt, as_mut)
 }
+
+/// Emits a sequence of `let id = expr;` from a vector of `[(id, expr), ...]`.
+///
+/// This doesn't do anything fancy, it's just to make sure the way that history
+/// expressions are treated in contract checks and replaces is uniform.
+pub fn create_history_expr_bindings(history_expressions: &[(Ident, Expr)]) -> TokenStream2 {
+    let history_variables = history_expressions.iter().map(|(v, _)| v);
```

Comment:
Maybe this should be
```
let (history_variables, history_exprs) = history_expressions.iter().unzip();
```
instead of creating two iters

Reply:
`unzip` requires creation of two intermediate vectors though because it doesn't split into an iterator but has to be collected.